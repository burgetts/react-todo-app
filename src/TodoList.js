import React, { useState } from 'react';
import Todo from './Todo'
import NewTodoForm from './NewTodoForm'
import { v4 as uuid } from 'uuid';

const TodoList = () => {
    const [todos, setTodos] = useState([])

    const addTodo = (newTodo) => {
        const newTodos = [...todos, {...newTodo, id: uuid()}]
        setTodos(newTodos)
    }
    const deleteTodo = (idToDelete) => {
        const newTodos = todos.filter(todo => todo.id !== idToDelete)
        setTodos(newTodos)
    }
    return (
        <>
            <h2> Add New Todo </h2>
            <NewTodoForm addTodo = {addTodo} />
            <h2> To Do List </h2>
            <ul>
                {todos.map(todo => <Todo id={todo.id} key={todo.id} text={todo.text} deleteTodo = {deleteTodo} />)}
            </ul>
        </>
    )
}

export default TodoList;