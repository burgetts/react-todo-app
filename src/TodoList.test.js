import React from 'react';
import {render, fireEvent} from '@testing-library/react'
import TodoList from './TodoList'

it('renders without crashing', function () {
    render(<TodoList  />)
})

it('matches snapshot', function () {
    const {asFragment} = render(<TodoList />)
    expect(asFragment()).toMatchSnapshot()
})

it('adds to do', function() {
    const {getByLabelText, getByText, queryByText} = render(<TodoList />)
    const input = getByLabelText('To do:')
    const btn = getByText('Add')

    expect(queryByText('laundry')).not.toBeInTheDocument()
    fireEvent.change(input, {target: {value: 'laundry'}})
    fireEvent.click(btn)

    expect(getByText('laundry')).toBeInTheDocument()
})

it('deletes to do', function() {
    const {getByLabelText, getByText, queryByText} = render(<TodoList />)
    const input = getByLabelText('To do:')
    const btn = getByText('Add')

    fireEvent.change(input, {target: {value: 'laundry'}})
    fireEvent.click(btn)

    expect(getByText('laundry')).toBeInTheDocument()

    const deleteBtn = getByText('‚ùå')
    fireEvent.click(deleteBtn)
    expect(queryByText('laundry')).not.toBeInTheDocument()
})