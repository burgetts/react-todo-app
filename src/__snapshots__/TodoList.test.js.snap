// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h2>
     Add New Todo 
  </h2>
  <div>
    <form>
      <label
        for="text"
      >
        To do:
      </label>
      <input
        id="text"
        name="text"
        placeholder="What do you need to do?"
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
  </div>
  <h2>
     To Do List 
  </h2>
  <ul />
</DocumentFragment>
`;
